File: frontend/src/api/comparison.ts
import { apiGet } from "./client";

export type PhaseLabelStats = {
  phase: string;
  label: number;
  labelName: string;
  voxelCount: number;
};

export type PhasePairDice = {
  phaseA: string;
  phaseB: string;
  label: number;
  labelName: string;
  dice: number;
};

export type ComparisonData = {
  caseId: string;
  phases: string[];
  labelStats: PhaseLabelStats[];
  diceScores: PhasePairDice[];
};

export function fetchPhaseComparison(caseId: string): Promise<ComparisonData> {
  return apiGet<ComparisonData>(`/api/cases/${caseId}/compare`);
}

File: backend/app/schemas/comparison.py
from pydantic import BaseModel


class PhaseLabelStats(BaseModel):
    phase: str
    label: int
    labelName: str
    voxelCount: int


class PhasePairDice(BaseModel):
    phaseA: str
    phaseB: str
    label: int
    labelName: str
    dice: float


class ComparisonResponse(BaseModel):
    caseId: str
    phases: list[str]
    labelStats: list[PhaseLabelStats]
    diceScores: list[PhasePairDice]
